langserver/completion.go

```diff
@@ -46,11 +47,20 @@ func (h *LangHandler) handleTextDocumentCompletion(ctx context.Context, conn jso
 		return nil, fmt.Errorf("invalid position: %s:%d:%d (%s)", filename, params.Position.Line, params.Position.Character, why)
 	}
 
-	ca, rangelen := gocode.AutoComplete(contents, filename, offset)
-	citems := make([]lsp.CompletionItem, len(ca))
-	for i, it := range ca {
+	ac, err := gocode.AutoComplete(&gocode.AutoCompleteRequest{
+		Filename: filename,
+		Data:     contents,
+		Cursor:   offset,
+		Source:   !h.config.UseBinaryPkgCache,
+		Context:  gbimporter.PackContext(h.BuildContext(ctx)),
```

Comment:
@keegancsmith Thanks for review. 

As I'm not using gocode daemon (not starting any net listener) I'm setting build context here. In current master https://github.com/sourcegraph/go-langserver/blob/master/langserver/internal/gocode/export.go#L17 is used only in unit tests. 
There are failing integration tests since this gocode fork returns different suggestions for some cases.

Reply:
Is it possible to update those tests instead of just commenting them out?

Ok great about setting the buildcontext where you set it, that actually makes a lot of sense.